packages/eslint-plugin/src/rules/no-implied-eval.ts

```diff
@@ -0,0 +1,141 @@
+import * as ts from 'typescript';
+import {
+  TSESTree,
+  AST_NODE_TYPES,
+} from '@typescript-eslint/experimental-utils';
+import * as util from '../util';
+
+const FUNCTION_CONSTRUCTOR = 'Function';
+const EVAL_LIKE_METHODS = new Set([
+  'setImmediate',
+  'setInterval',
+  'setTimeout',
+  'execScript',
```

Comment:
sorry to be a pain.
But `Function('console.log(1)')` is a function call.
```suggestion
  'execScript',
  FUNCTION_CONSTRUCTOR,
```
Could you add it to this set, just for completeness.

Reply:
You are right, `Function()` is the `CallExpression`, however, `CallExpression` and `NewExpression` have the same `callee` and it used to report the error `noFunctionConstructor`. It was done because it doesn't make sense to check `Function` arguments., `Function` doesn't provide any safe options to use arguments, like it does setTimeout, where we can use a function as the first argument.